enterprise/backend/src/metabase_enterprise/serialization/api.clj

```diff
@@ -0,0 +1,237 @@
+  (when-not (.exists src)
+    (throw (ex-info (format "Path is not readable or does not exist: %s" src)
+                    {:path src})))
+  (let [prefix (.getPath (.getParentFile src))]
+    (with-open [tar (-> (io/output-stream dst)
+                        (GzipCompressorOutputStream. (doto (GzipParameters.)
+                                                       (.setModificationTime (System/currentTimeMillis))))
+                        (TarArchiveOutputStream. 512 "UTF-8"))]
+      (.setLongFileMode tar TarArchiveOutputStream/LONGFILE_POSIX)
+
+      (doseq [^File f (file-seq src)
+              :let    [path-in-tar (subs (.getPath f) (count prefix))
+                       entry (TarArchiveEntry. f path-in-tar)]]
+        (.putArchiveEntry tar entry)
+        (when (.isFile f)
+          (with-open [s (io/input-stream f)]
+            (io/copy s tar)))
+        (.closeArchiveEntry tar)))
+    dst))
+
+(defn- entries [^TarArchiveInputStream tar]
+  (lazy-seq
+   (when-let [entry (.getNextEntry tar)]
+     (cons entry (entries tar)))))
+
+(defn- uncompress-tgz [^File archive ^File dst]
+  (let [dir-name (atom nil)]
+    (with-open [tar (-> (io/input-stream archive)
+                        (GzipCompressorInputStream.)
+                        (TarArchiveInputStream.))]
+      (doseq [^TarArchiveEntry e (entries tar)]
+        (when-not @dir-name
+          (reset! dir-name (.getName e)))
+        (let [f (io/file dst (.getName e))]
+          (if (.isFile e)
+            (io/copy tar f)
+            (.mkdirs f)))))
+    @dir-name))
+
+;;; Logging
+
+(defn- find-logger-layout
+  "Find any logger with a specified layout"
+  [^LoggerConfig logger]
+  (when logger
+    (or (first (keep #(.getLayout ^AbstractAppender (val %)) (.getAppenders logger)))
+        (recur (.getParent logger)))))
+
+(defn- make-logger ^AutoCloseable [ns ^File f]
+  (let [config        (logger/configuration)
+        parent-logger (logger/effective-ns-logger ns)
+        appender      (.build
+                       (doto (FileAppender/newBuilder)
+                         (.withName (.getPath f))
+                         (.withFileName (.getPath f))
+                         (.withLayout (find-logger-layout parent-logger))))
+        logger        (LoggerConfig. (logger/logger-name ns) Level/INFO true)]
+    (.start appender)
+    (.addAppender config appender)
+    (.addAppender logger appender (.getLevel logger) nil)
+    (.addLogger config (.getName logger) logger)
+    (.updateLoggers (logger/context))
+
+    (reify AutoCloseable
+      (close [_]
+        (let [^AbstractConfiguration config (logger/configuration)]
+          (.removeLogger config (.getName logger))
+          (.stop appender)
+          ;; this method is only present in AbstractConfiguration
+          (.removeAppender config (.getName appender))
+          (.updateLoggers (logger/context)))))))
+
+;;; Logic
+
+(defn- serialize&pack ^File [opts]
+  ;; TODO: how to get time in local timezone?
```

Comment:
> TODO: how to get time in local timezone?

`(t/local-date-time)` should do the trick?

Reply:
Nope, still formats result in UTC. :\